bundles/com.espressif.idf.core/src/com/espressif/idf/core/build/EditorOpenListener.java

```diff
@@ -0,0 +1,50 @@
+/*******************************************************************************
+ * Copyright 2023-2024 Espressif Systems (Shanghai) PTE LTD. All rights reserved.
+ * Use is subject to license terms.
+ *******************************************************************************/
+package com.espressif.idf.core.build;
+
+import org.eclipse.core.resources.IFile;
+import org.eclipse.ui.IEditorInput;
+import org.eclipse.ui.IEditorReference;
+import org.eclipse.ui.IPartListener2;
+import org.eclipse.ui.IWorkbenchPartReference;
+
+import com.espressif.idf.core.logging.Logger;
+
+/**
+ * Editor opener listener to check for the files and help run the marker remover job
+ * @see MarkerCleanupJob
+ * @author Ali Azam Rana
+ *
+ */
+public class EditorOpenListener implements IPartListener2
+{
+	@Override
+	public void partOpened(IWorkbenchPartReference partRef)
+	{
+		try
+		{
+			if (partRef instanceof IEditorReference)
+			{
+				IEditorReference editorRef = (IEditorReference) partRef;
+				if (editorRef.getEditorInput() instanceof IEditorInput)
+				{
+					IEditorInput fileInput = editorRef.getEditorInput();
+					IFile file = fileInput.getAdapter(IFile.class);
+					String filePath = file.getFullPath().toString();
+					if (filePath.contains("/ide/esp_idf_components/")) //$NON-NLS-1$
```

Comment:
is it possible to have the path separated by `\` here?

Reply:
I think the getFullPath will return the path with `/` instead of `\` for all operating systems so its better to use that instead of `\`